source/AI.cpp

```diff
@@ -1081,8 +1081,12 @@ void AI::MoveEscort(Ship &ship, Command &command) const
 		Refuel(ship, command);
 	else if(!parentIsHere && !isStaying)
 	{
-		// Check whether the ship has a target system and is able to jump to it.
+		// Check whether the ship has a target system and is able to jump to it,
+		// and that the targeted stellar object can be landed on.
 		bool hasJump = (ship.GetTargetSystem() && ship.JumpFuel(ship.GetTargetSystem()));
+		if(ship.GetTargetStellar() && (!ship.GetTargetStellar()->GetPlanet()
+				|| !(ship.GetTargetStellar()->GetPlanet() && ship.GetTargetStellar()->GetPlanet()->CanLand(ship))))
+			ship.SetTargetStellar(nullptr);
```

Comment:
The second check of `ship.GetTargetStellar()->GetPlanet()` is redundant. This could simply be done with:

```
ship.GetTargetStellar()
&& !(ship.GetTargetStellar()->GetPlanet() && ship.GetTargetStellar()->GetPlanet()->CanLand(ship))
```